src/utils/dependency-utils.ts

```diff
@@ -0,0 +1,676 @@
+        dependency,
+        reason: 'No license information available'
+      });
+      continue;
+    }
+    
+    // Check each license for the dependency
+    for (const license of dependency.licenses) {
+      // Try normalized license lookup first for better matching
+      const category = getLicenseCategoryNormalized(license);
+
+      if (!category) {
+        // Unknown license even after normalization
+        issues.push({
+          dependency,
+          reason: `Unknown license '${license}' - requires manual review`
+        });
+      } else if (category === LicenseCategory.X) {
+        // Category X - prohibited
+        issues.push({
+          dependency,
+          reason: `License '${license}' is in Category X (prohibited)`
+        });
+      } else if (category === LicenseCategory.B) {
+        // Category B - requires documentation
+        if (!isDocumentedInReadme(dependency, readmeContent)) {
+          // Check for special exceptions
+          if (isSpecialException(dependency.name)) {
+            // Special exception but still needs documentation
+            issues.push({
+              dependency,
+              reason: `Category B license '${license}' not documented in README (special exception: ${dependency.name})`
+            });
+          } else {
+            issues.push({
+              dependency,
+              reason: `Category B license '${license}' not documented in README`
+            });
+          }
+        }
+      }
+      // Category A licenses are automatically compliant
+    }
+  }
+  
+  return {
+    compliant: issues.length === 0,
+    issues
+  };
+}
+
+/**
+ * Check if a Maven project exists in the repository
+ * @param repoPath - Path to repository
+ * @returns Promise resolving to true if Maven project exists
+ */
+async function hasMavenProject(repoPath: string): Promise<boolean> {
+  if (!isNonEmptyString(repoPath)) {
+    return false;
+  }
+  return MAVEN_BUILD_FILES.some(file => fs.existsSync(path.join(repoPath, file)));
+}
+
+/**
+ * Check if a Gradle project exists in the repository
+ * @param repoPath - Path to repository
+ * @returns Promise resolving to true if Gradle project exists
+ */
+async function hasGradleProject(repoPath: string): Promise<boolean> {
+  if (!isNonEmptyString(repoPath)) {
+    return false;
+  }
+  return GRADLE_BUILD_FILES.some(file => fs.existsSync(path.join(repoPath, file)));
+}
+
+/**
+ * Extract dependencies from Maven project using license plugin
+ * @param repoPath - Path to Maven project
+ * @returns Promise resolving to array of dependencies
+ */
+async function getMavenDependencies(repoPath: string): Promise<Dependency[]> {
+  if (!isNonEmptyString(repoPath)) {
+    return [];
+  }
+
+  try {
+    // Execute Maven license plugin to generate third-party report
+    const { stdout } = await execAsync(
+      'mvn license:add-third-party -Dlicense.outputDirectory=target/licenses -Dlicense.includeTransitiveDependencies=false',
```

Comment:
> @maccabeelevine I hate to say this, and we should discuss, but doesn't includeTransitiveDependencies need to be true? And if so, the Dependency type probably has to track its ancestry.

Yes, lets discuss. I think we should leave as is. It would be holding FOLIO developers responsible for direct dependencies added to the project. They don't have control over that. Still, lawyers may not care if it is a transitive dependency and CAN take legal action.